### PR TITLE
Added jshint check, npm test, TravisCI config, and cleaned up some code.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by jonparrott** [comment]
Are you still planning to do a slightly refactor of `background.subscribe`?

**Comment by jmdobry** [comment]
Not right now

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+sudo: false
+language: node_js
+node_js:
+- "stable"
```
Comment (stephenplusplus) [inline .travis.yml]: Might be good to test 0.12 as well. Stable means 4.1 or 5 now.

**Comment by jonparrott** [comment]
We might as well go ahead and do it. It doesn't seem too big of a change.

**Comment by JustinBeckwith** [comment]
LGTM

File: .travis.yml (hunk 1 above)
Comment (jmdobry) [inline .travis.yml]: done

**Comment by jmdobry** [comment]
Merge?

**Comment by jonparrott** [comment]
I requested the refactoring to the `background` module, but I can give a stab at that myself. 

I'll integrate these changes then close this PR.

**Comment by jmdobry** [comment]
I quieted my own dislike for the current implementation of the `background` module, I'm not particular concerned about it anymore.

**Comment by stephenplusplus** [comment]
If it helps, `topic.get({ autoCreate: true }, function(err, topic) {})` is coming soon to gcloud :)

(`get()` (with an optional `{ autoCreate: bool }`), `create()`, and `exists()` will be available for all of our objects)

**Comment by jonparrott** [comment]
Done. This has all been integrated.

**Pull request closed without merging**